catalog/internal/catalog/performance_metrics.go

```diff
@@ -96,3 +122,362 @@ func LoadPerformanceMetricsData(path []string, modelRepo dbmodels.CatalogModelRe
 	glog.Infof("Successfully processed %d model directories and loaded %d models into database", processedCount, len(loadedModels))
 	return loadedModels, nil
 }
+
+// processModelDirectory processes a single model directory containing metadata.json and metric files
+func processModelDirectory(dirPath string, modelRepo dbmodels.CatalogModelRepository, metricsArtifactRepo dbmodels.CatalogMetricsArtifactRepository, modelTypeID int32, metricsArtifactTypeID int32, loadedModels map[string]*model.CatalogModel) error {
```

Comment:
Note; `metricsArtifactRepo` and `metricsArtifactTypeID` will be used in a subsequent PR which processes the metrics